File: game_server/ClientProtocol/GS2GC/p002_InitOp/GS2GC_002_061_RetShieldInit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p002_InitOp
{

/// <summary>
/// 屏蔽数据组件初始化
/// </summary>
public class GS2GC_002_061_RetShieldInit : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 已屏蔽玩家CID列表
/// </summary>
private List<long> shieldCidList;


public GS2GC_002_061_RetShieldInit() {
	shieldCidList = new List<long>();
}

public GS2GC_002_061_RetShieldInit(
	List<long> _shieldCidList
) {	shieldCidList = _shieldCidList;
}

public byte getMainOrder() { return (byte)2; }

public byte getSubOrder() { return (byte)61; }

/// <summary>
/// 已屏蔽玩家CID列表
/// </summary>
public List<long> getShieldCidList() { return shieldCidList; }
/// <summary>
/// 已屏蔽玩家CID列表
/// </summary>
public void addShieldCidList(long _shieldCidList) { shieldCidList.Add(_shieldCidList); }


public int GetBufSize() {
	int _size = 0;
	_size += 2 + (shieldCidList.Count * 8);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (shieldCidList.Count * 8);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _shieldCidListCount = _buf.getShort();
	for(int _i = 0; _i < _shieldCidListCount; _i++) { 
		long _shieldCidList = (long)0;
		_shieldCidList = _buf.getLong();
		shieldCidList.Add(_shieldCidList);
	}
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putShort((short)shieldCidList.Count);
	for(int _i = 0; _i < shieldCidList.Count; _i++) { 
		_buf.putLong(shieldCidList[_i]);
	}
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)2);
	_buf.put((byte)61);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)2);
	_recBuf.put((byte)61);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("shieldCidList").Append(":").Append(shieldCidList.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ServerProtocol/src/GS2GC/p002_InitOp/GS2GC_002_061_RetShieldInit.java
package GS2GC.p002_InitOp;

import java.nio.ByteBuffer;
/*********
 * 屏蔽数据组件初始化
 **/
public class GS2GC_002_061_RetShieldInit implements ALBasicProtocolPack._IALProtocolStructure {
/** 已屏蔽玩家CID列表 */
private java.util.ArrayList<Long> shieldCidList;


public GS2GC_002_061_RetShieldInit() {
	shieldCidList = new java.util.ArrayList<Long>();
}

public GS2GC_002_061_RetShieldInit(
	 java.util.ArrayList<Long> _shieldCidList
) {	shieldCidList = _shieldCidList;
}

public final byte getMainOrder() { return (byte)2; }

public final byte getSubOrder() { return (byte)61; }

/** 已屏蔽玩家CID列表 */
public java.util.ArrayList<Long> getShieldCidList() { return shieldCidList; }
/** 已屏蔽玩家CID列表 */
public void addShieldCidList(long _shieldCidList) { shieldCidList.add(_shieldCidList); }


public final int GetBufSize() {
	int _size = 0;
	_size += 2 + (shieldCidList.size() * 8);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (shieldCidList.size() * 8);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _shieldCidListCount = _buf.getShort();
	for(int _i = 0; _i < _shieldCidListCount; _i++) { 
		long _shieldCidList = (long)0;
		if(_buf.remaining() > 0) _shieldCidList = _buf.getLong();
		shieldCidList.add(_shieldCidList);
	}
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putShort((short)shieldCidList.size());
	for(int _i = 0; _i < shieldCidList.size(); _i++) { 
		_buf.putLong(shieldCidList.get(_i));
	}
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)2);
	_buf.put((byte)61);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)2);
	_recBuf.put((byte)61);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

